graphql_compiler/compiler/ir_lowering_common/location_renaming.py

```diff
@@ -43,10 +52,12 @@ def translate_potential_location(location_translations, potential_location):
         return potential_location
 
 
-def make_location_rewriter_visitor_fn(location_translations):
+def make_location_rewriter_visitor_fn(
+    location_translations: Dict[Type[BaseLocation], Type[BaseLocation]]
```

Comment:
I think this is incorrect. The dict does not map types to types, it maps values of that type to values of the other type. The `Type` here means that things like `BaseLocation` or `Location` should be put into this dict, not their instances (note I didn't use parens next to the names).

Also, as above, `location_translations` can only hold `Location`-typed keys and values, though there was no way to know this from this function or module alone.
```suggestion
    location_translations: Dict[Location, Location]
```